### ISSUE TITLE
We should clean up `package.json` from unnecessary fields before we publish `handsontable` on NPM

### ISSUE THREAD

**Issue by swistach** [handsontable/handsontable#7723]
### Description
Accordingly to the recent changes to introduce monorepo and modularisation, our `package.json` for dev env grew with additional fields such as:
- `sideEffects` - necessary for the end npm package, fortunately, supports wildcards in paths
- `exports` - necessary for end npm package, but doesn't support wildcards in paths, that means if we add, remove or rename plugins/editors/cell types/validators/i18n/languages/renderers, we have to remember to change that list manually,
- `workspaces` - necessary only for development, but in the end, npm package might [block CI/CD pipelines](https://forum.handsontable.com/t/yarn-throws-a-warning-when-installing-package/5140) if somebody uses `yarn` due to `workspaces` are allowed only for private packages: https://yarnpkg.com/features/workspaces/#how-to-declare-a-worktree

Problems we should solve:
- auto-generating `exports` - based on wildcard paths with automatic tests of `tmp/package.json` against non-existing references in files structure of the temporary directory,
- cleanup the `tmp/package.json` file from unnecessary fields such as `workspaces`, `devDependencies`, `scripts` and others if they are not for production with automatic tests to verify the final content,

### Steps to reproduce workspaces problem
Prerequisites: https://yarnpkg.com/getting-started/install
1. Create a new directory and run `yarn init` inside
2. Run `yarn add handsontable` or `yarn add handsontable@8.3.2`
You will see the following notification:
```
warning Workspaces can only be enabled in private projects.
```

### Steps to reproduce exports problem
1. Add a new renderer into `/src/renderers/newRenderer`
2. Run `npm build`
3. Build npm package from `/tmp`
4. Try to import your component from `handsontable/renderers/newRenderer`

Webpack will not recognise that path because it's not in exports.

### Your environment
* Handsontable version: 8.3.2
* yarn: 1.22.10
* npm: 7.7.6
* node: 15.13.0

Inform [forum](https://forum.handsontable.com/t/yarn-throws-a-warning-when-installing-package/5140)
Inform ZEN#26347

**Comment by aninde** [comment]
Fixed with v8.4.0.

**Comment by AMBudnik** [comment]
I'm glad to see this changed merged to Handsontable [v8.4.0](https://github.com/handsontable/handsontable/issues/7998).
